manila/share/drivers/vastdata/__init__.py

```diff
@@ -334,41 +219,72 @@ def shrink_share(self, share, new_size, share_server=None):
 
     def create_snapshot(self, context, snapshot, share_server):
         """Is called to create snapshot."""
-        path = self._to_volume_path(dict(id=snapshot['share_instance_id']))
-        LOG.info("Creating snapshot for %s", path)
-        snapshot = self.vms_session.post("snapshots", data=dict(
-            path=path,
-            name=snapshot['name'],
-        ))
+        path = self._to_volume_path(snapshot['share_instance_id'])
+        self.rest.snapshots.create(path=path, name=snapshot['name'])
 
     def delete_snapshot(self, context, snapshot, share_server):
         """Is called to remove share."""
-        name = snapshot['name']
-        snapshots = self.vms_session.snapshots(name=name)
-        assert len(snapshots) == 1, "Too many snapshots with name {!r}".format(name)
-        self.vms_session.delete("snapshots/{}".format(snapshots[0].id))
+        self.rest.snapshots.delete(name=snapshot["name"])
 
     def get_network_allocations_number(self):
         return 0
 
+    def _resize_share(self, share, new_size):
+        share_id = share['id']
+        quota = self.rest.quotas.one(name=share_id)
+        if not quota:
+            raise exception.ShareNotFound(reason="Share not found", share_id=share_id)
+
+        requested_capacity = new_size * units.Gi
+        self.rest.quotas.update(quota.id, hard_limit=requested_capacity)
+
+    def _delete_data_from_share(self, share_id):
+        src = self._to_volume_path(share_id)
+        LOG.info(f"Attempting trash API to delete '{src}'")
+        try:
+            self.rest.folders.delete(path=src)
+            return
+        except OperationNotSupported as exc:
+                LOG.debug(f"Trash API not available {exc}")
+
+        LOG.info(f"Use local mounting to delete {src}")
+
+        @contextmanager
+        def _mounted_root():
+            vip = random.choice(self.rest.vip_pools.vips(pool_name=self._vippool_name))
+            mp = f"/tmp/manila/{vip}"
+            utils.execute("mkdir", "-p", mp)
+            utils.execute("mount", "-t", "nfs", f"{vip}:{self._root_export}", mp, run_as_root=True)
+            try:
+                yield mp
+            finally:
+                utils.execute("umount", mp, run_as_root=True)
+                utils.execute("rmdir", mp, run_as_root=True)
+
+        with _mounted_root() as mount:
+            src = self._to_volume_path(share_id, mount)
+            dst = f"{mount}/deleted/{share_id}"
+            utils.execute("mkdir", "-p", f"{mount}/deleted", run_as_root=True)
+            utils.execute("mv", src, dst, run_as_root=True)
```

Comment:
let's drop support for local-mount deletion. we have no legacy to keep compatibility with here

Reply:
Done